sleap_nn/inference/inference.py

```diff
@@ -229,7 +418,7 @@ def forward(self, batch: Dict[str, torch.Tensor]) -> Dict[str, np.array]:
                     batch_peaks = torch.cat(
                         [
                             batch_peaks,
-                            torch.full((max_inst - num_inst, nodes, 2), torch.nan),
+                            torch.full((abs(max_inst - num_inst), nodes, 2), torch.nan),
```

Comment:
Why abs?? What if `num_inst > max_inst`?

Reply:
yes I need to add a case when the user doesn't provide max_instances attribute. The above code only works for the case max_instances in not None.

Reply:
Removed `max_instances` dependency and made it Optional. If max_instances is provided, then we set the limit, else all the predicted centroids would be processed